test/controllers/albums_controller_test.rb

```diff
@@ -0,0 +1,58 @@
+require 'test_helper'
+
+class AlbumsControllerTest < ActionController::TestCase
+  test "should get index" do
+    get :index
+    assert_response :success
+    assert_template :index
+  end
+
+  test "should get show" do
+    get :show, {id: albums(:two).id}
+    assert_response :success
+    assert_template :show
+  end
+
+  test "should get new" do
+    get :new
+    assert_response :success
+    assert_template :new
+  end
+
+  test "Creating an Album changes number of Albums" do
+    assert_difference('Album.count', 1) do
+      post_params = {album: {title: "A nice album"}}
+      post :create, post_params
+    end
+    assert_response :redirect
+  end
+
+  test "should get edit" do
+    get :edit, {id: 1}
+    assert_response :success
+    assert_template :edit
+  end
+
+  test "should update existing album" do
+    patch :update, id: albums(:one).id, album: { title: "title", artist: "artists"}
+    assert_response :redirect
+  end
```

Comment:
We could also test that updating an album does not change the number of albums in the database.